### ISSUE TITLE
Incorrect feed ibash.de

### ISSUE THREAD

**Issue by 0x6d69636b** [Riduidel/rrss2imap#42]
Feed at url http://www.ibash.de/neueste-zitate.xml doesn't seems to be parseable

**Comment by Riduidel** [comment]
Sorry for the very late reply.
Do you have any error log ?

**Comment by 0x6d69636b** [comment]
No worries :-)

I just ran it again, here parts of the output. Does this help for the analysis?

```
$ /opt/rrss2imap/rrss2imap -V  
rrss2imap 0.2.42

$ /opt/rrss2imap/rrss2imap run
...
[2020-10-01 18:21:43.149851 +02:00] INFO [rrss2imap::feed] src/feed.rs:87: Reading feed 82/82 fro
m http://www.ibash.de/neueste-zitate.xml
[2020-10-01 18:21:43.274170 +02:00] INFO [rrss2imap::feed_reader] src/feed_reader.rs:65: Feed dat
e is 2020-10-01 16:21:43 while previous read date is 1970-01-01 00:00:00
[2020-10-01 18:21:43.274480 +02:00] ERROR [rrss2imap::feed_reader] src/feed_reader.rs:214: feed i
tem Some("http://www.ibash.de/zitat_60546.html") date can't be parsed, as it doesn't have neither
 pub_date nor dc:pub_date. We will replace it with feed date if possible
[2020-10-01 18:21:43.274773 +02:00] ERROR [rrss2imap::feed_reader] src/feed_reader.rs:214: feed i
tem Some("http://www.ibash.de/zitat_60544.html") date can't be parsed, as it doesn't have neither
 pub_date nor dc:pub_date. We will replace it with feed date if possible
[2020-10-01 18:21:43.275045 +02:00] ERROR [rrss2imap::feed_reader] src/feed_reader.rs:214: feed i
tem Some("http://www.ibash.de/zitat_60541.html") date can't be parsed, as it doesn't have neither
 pub_date nor dc:pub_date. We will replace it with feed date if possible
[2020-10-01 18:21:43.275325 +02:00] ERROR [rrss2imap::feed_reader] src/feed_reader.rs:214: feed i
tem Some("http://www.ibash.de/zitat_60539.html") date can't be parsed, as it doesn't have neither
 pub_date nor dc:pub_date. We will replace it with feed date if possible
```

**Comment by Z01ndels** [comment]
Thanks! I'll try to look at that next week

**Comment by Riduidel** [comment]
Thanks! I'll try to take a look at that next wekk (but I'm quite confident in the fact the source feed doesn't send any date info for its entries.

**Comment by 0x6d69636b** [comment]
Would it be possible to process the feed in despite of missing timestamps and just use the current time as date/time?

**Comment by Riduidel** [comment]
Yes, obviously, I will try that. But I suspect there is some kind of incoherent behaviour between date lib on x86 and on arm (because I've got some feeds which seems to have correct dates when I test on my machine, but behave strangely on my Raspberry)

**Comment by Riduidel** [comment]
ok, so your feeddon't even have any date field set.
I've never thought I would meet such a case.
Let me rephrase that : as of today, all the feeds I read have dates, which I use as marker to get more recent entries. Without dates, I will have to rely upon entries ids and output all entries until I meet one I've already written.
I think I should first read the rss/atoms specs before doing something more stupid than usual.

**Comment by Riduidel** [comment]
Daaaaaamn!
It appear the `pubDate` is optional! How can I get only new entries on non-dated feeds ? This will be something quite savage.

**Comment by Riduidel** [comment]
ok, now I understand my error, I guess I'm ready to remove the code testing the feed entry date (see in `src/feed_reader.rs` function `Feed_Reader#filter_message(...)`)

**Comment by 0x6d69636b** [comment]
The RSS/Atom specification is (unfortunately) very liberal with optional fields like date and IDs. :( In case of this feed the link would be a ID attribute at best. But that would mean that the last link would have to be stored somewhere and then be compared on the next fetch. And it can differ from feed to feed which is the ID attribute now, unless of course a date is used. I hope you have a better/simpler idea than me?
